theseus_gui/src/store/instancesStore.js

```diff
@@ -0,0 +1,185 @@
+        },
+        {
+          id: 5,
+          name: 'SkyFactory 4',
+          description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit',
+          version: '1.12.2',
+          downloads: 1000,
+          trending: false,
+          img: 'https://cdn.modrinth.com/user/MpxzqsyW/eb0038489a55e7e7a188a5b50462f0b10dfc1613.jpeg',
+        },
+        {
+          id: 6,
+          name: 'RLCraft',
+          description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit',
+          version: '1.12.2',
+          downloads: 10000,
+          trending: false,
+          img: 'https://avatars1.githubusercontent.com/u/6166773?v=4',
+        },
+        {
+          id: 7,
+          name: 'Regrowth',
+          description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit',
+          version: '1.7.10',
+          downloads: 1000,
+          trending: false,
+          img: 'https://cdn.modrinth.com/data/ssUbhMkL/icon.png',
+        },
+        {
+          id: 8,
+          name: 'Birds',
+          description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit',
+          version: '1.15.2',
+          downloads: 9,
+          trending: false,
+          img: 'https://avatars.githubusercontent.com/u/83074853?v=4',
+        },
+        {
+          id: 9,
+          name: 'Dogs',
+          description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit',
+          version: '1.15.2',
+          downloads: 9,
+          trending: false,
+          img: 'https://cdn.modrinth.com/user/MpxzqsyW/eb0038489a55e7e7a188a5b50462f0b10dfc1613.jpeg',
+        },
+        {
+          id: 10,
+          name: 'Cats',
+          description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit',
+          version: '1.15.2',
+          downloads: 9,
+          trending: false,
+          img: 'https://cdn.modrinth.com/data/ssUbhMkL/icon.png',
+        },
+        {
+          id: 11,
+          name: 'Rabbits',
+          description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit',
+          version: '1.15.2',
+          downloads: 9,
+          trending: false,
+          img: 'https://avatars1.githubusercontent.com/u/6166773?v=4',
+        },
+      ]
+
+      this.fakeInstances = [...instances]
+    },
+    async searchInstances() {
+      const response = await ofetch(
+        `https://staging-api.modrinth.com/v2/search?query=${this.searchInput || ''}&offset=${
+          this.offset || 0
+        }`
+      )
+      this.instances = [...response.hits]
+      this.totalHits = response.total_hits
+      this.offset = response.offset
+      this.pageCount = Math.ceil(this.totalHits / 10)
+    },
+    setSearchInput(newInput) {
+      this.searchInput = newInput
+    },
+    setCurrentPage(newPage) {
+      this.currentPage = newPage
+
+      if (newPage === 1) this.offset = 0
+      else this.offset = this.currentPage * 10 - 10
+    },
+    setFilter(newFilter) {
+      this.filter = newFilter
+    },
```

Comment:
The search functionality shouldn't be inside stores IMO. It's fine to just have it in Javascript in the `script setup`. We're not going to use search functions anywhere else